src/bentoml/_internal/io_descriptors/pandas.py

```diff
@@ -527,12 +526,15 @@ async def from_http_request(self, request: Request) -> ext.PdDataFrame:
         _validate_serialization_format(serialization_format)
 
         obj = await request.body()
+        dtype = self._dtype if self._enforce_dtype else None
```

Comment:
this could be a breaking change for users using `from_sample` or setting `dtype` today 🤔

Reply:
I think this breaks spark, because even if enforce dtype is false we use the user `dtype` to determine output table dtype.

Reply:
@sauyon this is only in the from_http_request code path, does not affect the spark schema method I believe

Reply:
Ah, right, I thought this was in the constructor.